src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,48 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
+                Type payloadType = payload[i].GetType();
+                if(payloadType == typeof(string))
```

Comment:
You can write this like this to save on redundant casts:

```
switch (payload[i])
{
case String str:
...
       break;
case byte[] arr:
... 
      minimalJsonserializer(arr, sb);
...
      break;
default:
       ...
}    
```

Reply:
Thanks @jkotas.  I like that pattern much better.